payjoin/src/receive/mod.rs

```diff
@@ -288,6 +288,45 @@ pub trait Headers {
     fn get_header(&self, key: &str) -> Option<&str>;
 }
 
+#[cfg(feature = "v2")]
```

Comment:
this design i know is transient until the v2 file is created. If the commit just started with a smaller piece of the final design rather than jump to it it would be easier to review